src/rayleigh_quotient.jl

```diff
@@ -85,12 +88,35 @@ end
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_EIG)
     b = prob.b
     C = prob.C
-    Kb = I - b * b' / dot(b, b)
-    nlargest = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
     N = size(b, 1)
-    inds = map(!in(nlargest), 1:N)
-    J = I(N)[inds, :]
+    Kb = I - b * b' / dot(b, b)
+    _, inds_keep = _inds_to_keep_and_remove(b, N)
+    J = I(N)[inds_keep, :]
     augC = J * Kb * C
+    return _solve_homo_prob_and_normalize(augC, prob, alg)
+end
+
+function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_SPARSE)
+    b = prob.b
+    C = prob.C
+    N = size(b, 1)
+    inds_remove, inds_keep = _inds_to_keep_and_remove(b, N)
+    J = I(N)[inds_keep, :]
+    bk = first(b[inds_remove]) # only one element, but it seems like we can have b a matrix?
+    Ck = C[inds_remove, 1:N]
```

Comment:
Where does this 1:N come from?

Reply:
But I don't think it has to have size N? Let's change the test instead to use I(N).

Reply:
Yes, that sounds good. I'll make a separate PR with updated tests.